core/src/main/scala/io/udash/properties/seq/ZippedSeqProperty.scala

```diff
@@ -2,42 +2,63 @@ package io.udash.properties.seq
 
 import com.avsystem.commons._
 import io.udash.properties._
-import io.udash.properties.single.ReadableProperty
+import io.udash.properties.single.{CombinedProperty, ReadableProperty}
 import io.udash.utils.{CrossCollections, Registration}
 
-private[properties] abstract class ZippedSeqPropertyUtils[O] extends AbstractReadableSeqProperty[O, ReadableProperty[O]] {
-  override val id: PropertyId = PropertyCreator.newID()
-  override protected[properties] val parent: ReadableProperty[_] = null
-
-  protected var children: MBuffer[ReadableProperty[O]] = _
-  protected final val originListener: Patch[ReadableProperty[_]] => Unit =
-    (patch: Patch[ReadableProperty[_]]) => {
-      val idx = patch.idx
-      val els = children
-      val removed = CrossCollections.slice(els, patch.idx, els.length)
-      val added = updatedPart(idx)
-      if (added.nonEmpty || removed.nonEmpty) {
-        CrossCollections.replaceSeq(els, idx, removed.size, added)
-        val mappedPatch = Patch(patch.idx, removed.toSeq, added, patch.clearsProperty)
-        CallbackSequencer().queue(
-          s"${this.id.toString}:fireElementsListeners:${patch.hashCode()}",
-          () => structureListeners.foreach(_.apply(mappedPatch))
-        )
-        valueChanged()
-      }
+private[properties] abstract class ZippedSeqPropertyUtils[O](
+  sources: ISeq[ReadableSeqProperty[_, _ <: ReadableProperty[_]]]
+) extends AbstractReadableSeqProperty[O, ReadableProperty[O]] {
+
+  override final val id: PropertyId = PropertyCreator.newID()
+  override final protected[properties] def parent: ReadableProperty[_] = null
+
+  private final val children = CrossCollections.createArray[ReadableProperty[O]]
+  private final val sourceRegistrations = CrossCollections.createArray[Registration]
+  private final val childrenRegistrations = CrossCollections.createArray[Registration]
+
+  private val originStructureListener: Patch[ReadableProperty[_]] => Unit = { patch =>
+    val removed = CrossCollections.slice(children, patch.idx, children.length)
+    val added = updatedPart(patch.idx)
+    if (added.nonEmpty || removed.nonEmpty) {
+      CrossCollections.replaceSeq(children, patch.idx, removed.size, added)
+      val mappedPatch = Patch(patch.idx, removed.toSeq, added, patch.clearsProperty)
```

Comment:
`.toSeq` unneeded

Reply:
Do you have a moment to talk about Scala 2.13, sir?